packages/ruby/src/install-ruby.ts

```diff
@@ -47,12 +49,69 @@ function getLatestRubyVersion(): RubyVersion {
   return selection;
 }
 
+function resolveSystemRuby(): {
+  rubyPath: string;
+  gemPath: string;
+  major: number;
+  minor: number;
+} | null {
+  const rubyPath = which.sync('ruby', { nothrow: true }) as string | null;
+  const gemPath = which.sync('gem', { nothrow: true }) as string | null;
+  if (!rubyPath || !gemPath) return null;
+  const ver = spawnSync(rubyPath, ['-e', 'print RUBY_VERSION'], {
+    encoding: 'utf8',
+  });
+  if (ver.status !== 0 || !ver.stdout) return null;
+  const [mj, mn] = String(ver.stdout).trim().split('.');
+  const major = Number(mj) || 3;
+  const minor = Number(mn) || 3;
+  return { rubyPath, gemPath, major, minor };
+}
+
+function makeLocalRubyEnv({
+  major,
+  minor,
+  rubyPath,
+  gemPath,
+}: {
+  major: number;
+  minor: number;
+  rubyPath: string;
+  gemPath: string;
+}) {
+  const gemHome = join(tmpdir(), `vercel-ruby-${major}${minor}-${process.pid}`);
+  return {
+    major,
+    gemHome,
+    runtime: `ruby${major}.${minor}`,
```

Comment:
We should have this set to whatever we have in the Vercel build environment so that we're deploying consistently across the two. 

The way that I've thought about this for CLI outside of Vercel is we bring in your configuration setting from Vercel and locally you're responsible for setting your your environment equivalently.